presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/OrganizationJob.java

```diff
@@ -56,34 +57,39 @@ public OrganizationJob(OrganizationSet organizationSet, MetadataDao metadataDao,
     public void run()
     {
         try {
-            runJob(organizationSet.getTableId(), organizationSet.getBucketNumber(), organizationSet.getShards());
+            runJob(organizationSet.getTableId(), organizationSet.isTableSupportsDeltaDelete(), organizationSet.getBucketNumber(), organizationSet.getShardsMap());
         }
         catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
-    private void runJob(long tableId, OptionalInt bucketNumber, Set<UUID> shardUuids)
+    private void runJob(long tableId, boolean tableSupportsDeltaDelete, OptionalInt bucketNumber, Map<UUID, Optional<UUID>> shardUuidsMap)
             throws IOException
     {
         long transactionId = shardManager.beginTransaction();
         try {
-            runJob(transactionId, bucketNumber, tableId, shardUuids);
+            runJob(transactionId, tableSupportsDeltaDelete, bucketNumber, tableId, shardUuidsMap);
         }
         catch (Throwable e) {
             shardManager.rollbackTransaction(transactionId);
             throw e;
         }
     }
 
-    private void runJob(long transactionId, OptionalInt bucketNumber, long tableId, Set<UUID> shardUuids)
+    private void runJob(long transactionId, boolean tableSupportsDeltaDelete, OptionalInt bucketNumber, long tableId, Map<UUID, Optional<UUID>> shardUuidsMap)
             throws IOException
     {
+        log.info("Start compacting shards %s for table: %s", shardUuidsMap, tableId);
         TableMetadata metadata = getTableMetadata(tableId);
-        List<ShardInfo> newShards = performCompaction(transactionId, bucketNumber, shardUuids, metadata);
-        log.info("Compacted shards %s into %s", shardUuids, newShards.stream().map(ShardInfo::getShardUuid).collect(toList()));
-        // todo add logic in organization for delta
-        shardManager.replaceShardUuids(transactionId, tableId, metadata.getColumns(), null, newShards, OptionalLong.empty(), false);
+        List<ShardInfo> newShards = performCompaction(transactionId, tableSupportsDeltaDelete, bucketNumber, shardUuidsMap, metadata);
+        log.info("Compacted shards %s into %s for table: %s", shardUuidsMap, newShards.stream().map(ShardInfo::getShardUuid).collect(toList()), tableId);
+        if (tableSupportsDeltaDelete) {
+            shardManager.replaceShardUuids(transactionId, tableId, metadata.getColumns(), shardUuidsMap, newShards, OptionalLong.empty(), tableSupportsDeltaDelete);
```

Comment:
nit: Maybe we can rename the delta version of `replaceShardUuids()` to another name. Here IntelliJ is complaining about `tableSupportsDeltaDelete ` is always true when passing to this function.